src/Workspaces/Core/Portable/CodeActions/CodeAction.cs

```diff
@@ -52,6 +52,14 @@ public abstract class CodeAction
         /// </remarks>
         public virtual string? EquivalenceKey => null;
 
+        /// <summary>
+        /// Identifies the Code Action Provider which produced this Code Action.
+        /// </summary>
+        /// <remarks>
+        /// e.g. 'Add Await Code Action Provider'
+        /// </remarks>
+        internal virtual string? ProviderName => null;
```

Comment:
We may want to consider exposing this property as a public API - this is something that can be made part of the code action telemetry and we can resolve the long outstanding issue https://github.com/dotnet/roslyn/issues/4919. @allisonchou do you mind bringing it to the design meeting?